Packs/GitLab/Integrations/GitLab/GitLab.py

```diff
@@ -0,0 +1,1584 @@
+                               'Name': group.get('name'),
+                               'Path': group.get('path'),
+                               'Description': group.get('descrition', ''),
+                               'CreatedAt': group.get('created_at', ''),
+                               'Visibility': group.get('visibility', '')})
+    human_readable = tableToMarkdown(response_title, response_to_hr, removeNull=True, headers=headers)
+    return CommandResults(
+        outputs_prefix='GitLab.Group',
+        outputs_key_field='Id',
+        readable_output=human_readable,
+        outputs=response,
+        raw_response=response
+    )
+
+
+def issue_note_create_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Creates an issue note.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'issue_iid': the IID of the issue.
+            - 'body': Retrieve only issues with the given labels.
+            - 'confidential': If the thread is confidential.
+
+    Returns:
+        (CommandResults).
+    """
+    issue_iid = args.get('issue_iid', '')
+    body = args.get('body', '')
+    confidential = args.get('confidential')
+    response = client.issue_note_create_request(issue_iid, body, confidential)
+    human_readable_str = 'Issue note created successfully'
+    return CommandResults(
+        outputs_prefix='GitLab.IssueNote',
+        readable_output=human_readable_str,
+        outputs=response,
+        raw_response=response
+    )
+
+
+def issue_note_delete_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    deletes an issue note.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'issue_iid': the IID of the issue.
+            - 'note_id': the ID of the issue note.
+
+    Returns:
+        (CommandResults).
+    """
+    issue_iid = arg_to_number(args.get('issue_iid'))
+    note_id = arg_to_number(args.get('note_id'))
+    client.issue_note_delete_request(issue_iid, note_id)
+    human_readable_str = 'Issue note deleted successfully'
+    return CommandResults(
+        readable_output=human_readable_str
+    )
+
+
+def issue_note_update_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    updating an issue note.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'state': The state of the issue.
+            - 'labels': Retrieve only issues with the given labels.
+            - 'assignee_username': Retrieve issues by assignee username.
+
+    Returns:
+        (CommandResults).
+    """
+    issue_iid = args.get('issue_iid')
+    note_id = args.get('note_id')
+    body = args.get('body')
+    if not issue_iid or not note_id or not body:
```

Comment:
All of them required?

Reply:
yes

Reply:
Are they marked as required in the yml? if so, you don't need this check

Reply:
ok, I removed it